### ISSUE TITLE
Path-returned pages not regenerating

### ISSUE THREAD

**Issue by Timer** [vercel/next.js#14073]
I'm experiencing an issue with pages that *are* generated at build time are not revalidated.

For instance, this does not regenerate the page when changes are made in the CMS:

```ts
/**
 * Initial Props
 */

export const getStaticProps: GetStaticProps = async (ctx) => {
  const { params } = ctx
  if (!params) return { props: { product: undefined } }
  const variables = { handle: params.productSlug }
  const response = await request<Response>(productQuery, variables)
  const products = response?.allShopifyProduct

  const product = products && products.length ? products[0] : null
  return { props: { product }, unstable_revalidate: 1 }
}

/**
 * Static Paths
 */
const pageHandlesQuery = gql`
  query ProductHandlesQuery {
    allShopifyProduct {
      _id
      shopifyId
      handle
    }
  }
`

export const getStaticPaths: GetStaticPaths = async () => {
  const result = await request<Response>(pageHandlesQuery)
  const products = definitely(result?.allShopifyProduct)
  const paths = products.map((product) => ({
    params: { productSlug: product.handle ? product.handle : undefined },
  }))

  return {
    paths,
    fallback: true,
  }
}
```

But this does:

```ts
/**
 * Initial Props
 */

export const getStaticProps: GetStaticProps = async (ctx) => {
  const { params } = ctx
  if (!params) return { props: { product: undefined } }
  const variables = { handle: params.productSlug }
  const response = await request<Response>(productQuery, variables)
  const products = response?.allShopifyProduct

  const product = products && products.length ? products[0] : null
  return { props: { product }, unstable_revalidate: 1 }
}

/**
 * Static Paths
 */


export const getStaticPaths: GetStaticPaths = async () => {
  return {
    paths: [],
    fallback: true,
  }
}
```

It's the same behavior on the final example in this blog post:

https://arunoda.me/blog/what-is-nextjs-issg

Is this how the feature is supposed to work?

_Originally posted by @good-idea in https://github.com/vercel/next.js/discussions/12775#discussioncomment-24824_

**Comment by good-idea** [comment]
I'm not sure how the props are compared when revalidating -- in my case, the "product" prop likely had changes that were nested a few levels deep in the object, i.e. `product.images[0].altText` might have changed

**Comment by Timer** [comment]
We've confirmed this behavior works and cannot be reproduced with the instructions provided by @good-idea.

If you can provide a full reproduction, happy to take a look!

**Comment by good-idea** [comment]
Hey @Timer, thanks for looking into this, sorry to send you down a rabbit hole. I'll try re-implementing it in my project and see if I can recreate it.

**Comment by Timer** [comment]
Here is a separate bug you may want to track (being fixed today): #13900

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
